lib/helper/Playwright.js

```diff
@@ -2699,7 +2679,8 @@ async function getXPathForElement(elementHandle) {
 async function findElements(matcher, locator) {
   if (locator.react) return findReact(matcher, locator);
   locator = new Locator(locator, 'css');
-  return matcher.$$(buildLocatorString(locator));
+
+  return matcher.locator(buildLocatorString(locator)).all();
```

Comment:
It could be a huge issue for projects with dynamic rendering. 
[$$](https://playwright.dev/docs/api/class-elementhandle#element-handle-query-selector-all) from ElementHandle  awaits for all matching elements, but [.all()](https://playwright.dev/docs/api/class-locator#locator-all) from Locator not, it immediately returns whatever is present in the page. 

So as the function `findElements()` use as entry point for locating, it's dangerous to stay as is

Reply:
[locator.all()](https://playwright.dev/docs/api/class-locator#locator-all) does not wait for elements to match the locator, but without `.all()` waits

So to make selectors more stable better to use it as often as we can without  the`.all()` method

Reply:
I'm not sure I understand all details here but here is the plan:

@kobenguyent try to follow suggestions from @EgorBodnar 
when you think the fix is ready I will **publish a beta release from this branch** and we will ask community to test it. If everything is ok, I will merge it and make a proper release